### PR TITLE
aws-c-auth/0.6.0

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **aws-c-auth/0.6.0**

ref #6346
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by intelligide** [comment]
CCI doesn't contains the prebuilt package for aws-c-http/0.6.5 for Windows.

image
(https://conan.io/center/aws-c-http)

It seems that this is a build pipeline error.

**Comment by jgsogo** [comment]
I'm on it, but builds for `qt` entered before that one... so we need to wait a little bit.

**Comment by jgsogo** [comment]
Packages built. All those missing ones were really unexpected... https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/TapaholesList/237/summary.json

File: recipes/aws-c-auth/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (intelligide) [inline recipes/aws-c-auth/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

**Review by intelligide — CHANGES_REQUESTED** [review]

**Review by intelligide — DISMISSED** [review]

**Review by jgsogo — DISMISSED** [review]

**Comment by MartinDelille** [comment]
I added missing requirements after @intelligide [comments](https://github.com/conan-io/conan-center-index/pull/6715#discussion_r684641978).

These libraries are indeed reference in the source code: https://github.com/awslabs/aws-c-auth/blob/main/source/credentials_provider_x509.c#L10-L20

I'm not sure to understand why it is needed since it seems to work without and it may make bumping dependencies version more tricky.

I'm also wondering if there was a simple way to detect these dependencies.

**Review by jgsogo — APPROVED** [review]
It is not required (it builds and works) to add those transitive dependencies, but it is a good practice when the package uses them directly. It can happen that the requirement in some future version stops using that transitive dependency, but the library still needs it. Adding it as a direct requirement ensures that the recipe will keep working if that scenario happens.

For a _framework_ like `aws-c-xxxx` it's very likely that the dependency graph will only grow in complexity, but for other dependency graphs of unrelated libraries the situation has a higher probability.

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**